subsys/caf/modules/ble_state.c

```diff
@@ -269,19 +266,11 @@ static void security_changed(struct bt_conn *conn, bt_security_t level,
 	event->state = PEER_STATE_SECURED;
 	EVENT_SUBMIT(event);
 
-	struct bt_conn_info info;
-
-	err = bt_conn_get_info(conn, &info);
-	if (err) {
-		LOG_WRN("Cannot get conn info");
-	} else {
-		if (IS_ENABLED(CONFIG_BT_CENTRAL) &&
-		    (info.role == BT_CONN_ROLE_MASTER)) {
-			exchange_params.func = exchange_func;
-			err = bt_gatt_exchange_mtu(conn, &exchange_params);
-			if (err) {
-				LOG_ERR("MTU exchange failed");
-			}
+	if (IS_ENABLED(CONFIG_CAF_BLE_STATE_EXCHANGE_MTU)) {
+		exchange_params.func = exchange_func;
```

Comment:
don't we want to keep the role check?

Reply:
The Kconfig option depends on CONFIG_BT_GATT_CLIENT. Thanks to that the check if more generic (slave could be a GATT_CLIENT too in some applications). According to Zephyr docs the bt_gatt_exchange_mtu is `client procedure can be used to set the MTU to the maximum possible size the buffers can hold`.

For nrf_desktop usecase it makes no difference, since only dongle (BT Central) is a GATT Client.